src/content/app/tools/blast/components/blast-job-submit/BlastJobSubmit.tsx

```diff
@@ -89,13 +89,17 @@ const BlastJobSubmit = () => {
 export const createBlastSubmissionData = (
   blastFormData: BlastFormState
 ): PayloadParams => {
-  // labelling sequences with complely artificial identifiers
+  // labelling sequences with completely artificial identifiers
   // so that job ids in the response can be matched to individual combinations of sequences and genome ids
   const sequences = blastFormData.sequences.map((sequence, index) => ({
     id: index + 1,
     header: sequence.header,
     value: sequence.value
   }));
+  sequences.forEach((item) => {
+    if (!item.header) delete item.header;
+  });
```

Comment:
Yuck! Let's do this inside the map instead.

If `blastFormData.sequences` sometimes have the header field and sometimes don't (please check), then it's as easy as:

```
const sequences = blastFormData.sequences.map((sequence, index) => ({
  ...sequence,
  id: index + 1,
}));
``` 

Otherwise, we'll need to do something like this:
```
const sequences = blastFormData.sequences.map((sequence, index) => {
 const sequenceForSubmission: SubmittedSequence = {
  id: index + 1,
  value: sequence.value
 };

 if (sequence.header) {
  sequenceForSubmission.header = sequence.header;
 }

 return sequenceForSubmission;
});
```

Reply:
Turn out the first option works since `blastFormData.sequences` dont have the header field when its empty. But if it did i am curious to why my solution wasn't good. 
i have made the suggested change but just curious why not the delete option.

Reply:
>  But if it did i am curious to why my solution wasn't good. whats the difference between the delete and the map. Is it because we dont want to mutate the object?

It's probably a matter of style; but between (1) creating a header field and then deleting the header field if empty, and (2) not creating the header field if empty in the first place, the second option feels much more satisfying.